src/txmempool.cpp

```diff
@@ -737,7 +738,7 @@ void CTxMemPool::check(const CCoinsViewCache *pcoins) const
             waitingOnDependants.push_back(&(*it));
         else {
             CValidationState state;
-            assert(CheckInputs(tx, state, mempoolDuplicate, false, 0, false, NULL));
+            assert(Consensus::CheckTxInputs(tx, state, mempoolDuplicate, nSpendHeight));
```

Comment:
Thanks. 
So, yes, it seems these asserts were not complying with that rule. Neither before not after my change.
If there's an easy fix to add here, I'm all for it.
Should I launch an exception instead ?
It doesn't seem right the way CTxMemPool::check is written...
Maybe it's enough to rename the offending state local variables to dummyState ?